### PR TITLE
refactor: use separate forms in profile setting tabs

### PR DESCRIPTION
**Pull request by goga-m**
## Summary
-   [x] Isolate `General`, `Password` and `Export` settings into standalone form contexts and states,
-   [x] Cleanup Settings page component
-   [x] Fix issue with profile name not applying updated name after save
-   [x] Remove confirmation prompt modal from `Export` settings
-   [x] Apply theme after resetting profile settings
-   [x] Prevent from reloading settings page after save. Scroll up instead.

## Checklist

-   [x] My changes look good in both light AND dark mode
-   [x] The change is not hardcoded to a single network, but has multi-asset in mind
-   [x] I checked my changes for obvious issues, debug statements and commented code
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

**Comment by faustbrian** [comment]
Task linked: [CU-n7cghm [Settings] Remove Confirmation prompt from `Export` settings](https://app.clickup.com/t/n7cghm)

**Comment by faustbrian** [comment]
Task linked: [CU-n16cq4 [Setting] Profile name is not updated in form after changed and saved](https://app.clickup.com/t/n16cq4)

**Comment by faustbrian** [comment]
Task linked: [CU-n7dcgn [Settings] Theme is not applied after reset](https://app.clickup.com/t/n7dcgn)

File: src/domains/setting/pages/Settings/Settings.tsx [hunk 1]
```diff
@@ -1,96 +1,24 @@
...
-		const ActiveSettings = availableSettings[activeSettings];
-
-		return (
-			<ActiveSettings
-				formConfig={{ context: form, register, errors }}
-				onSuccess={handleSuccess}
-				onError={handleError}
-			/>
-		);
-	};
```
Comment (w3ea) [inline src/domains/setting/pages/Settings/Settings.tsx:85]: I think we can still use this function to render setting page dynamically
like this (tested)
```
const renderSettings = () => {
	const ActiveSettings = availableSettings[activeSettings];

	return (
		<ActiveSettings/>
	);
};
```

Comment (goga-m) [reply src/domains/setting/pages/Settings/Settings.tsx:85]: @w3ea I removed the renderSettings mostly for readability as it seemed more clear to know which component is rendering :hmhm: . The Settings component now renders the 3 tabs as: 
```jsx
export const Settings = () => {
	const activeProfile = useActiveProfile();
	const { menuItems, activeSettings, setActiveSettings } = useSettingsMenu();

	return (
		<Page
			profile={activeProfile}
			sidebar={<SideBar items={menuItems} activeItem={activeSettings} handleActiveItem={setActiveSettings} />}
		>
			<Section innerClassName="px-12">
				{activeSettings === "General" && <General />}
				{activeSettings === "Password" && <PasswordSettings />}
				{activeSettings === "Export" && <Export />}
			</Section>
		</Page>
	);
};

```

Comment (w3ea) [reply src/domains/setting/pages/Settings/Settings.tsx:85]: yeah, i got that
but for adding a new page we need to change two place
[here ](https://github.com/ArkEcosystem/desktop-wallet/pull/3969/files#diff-fd2e0e758c3d350d8b2e7baa8f38e25cc6bc7f18d92151df18ba3942627cc29cR8-R24) and [here](https://github.com/ArkEcosystem/desktop-wallet/pull/3969/files#diff-464370bbc0ddbe87686580ee018bc24e908d5083e34ac34218f9aa1524992a7bR18-R20)
it's totally ok, it was just a suggestion

**Review by alesmit — APPROVED** [review]

**Comment by w3ea** [comment]
@goga-m FYI
i can still see focus issue :hmhm:

https://user-images.githubusercontent.com/24996341/123781641-c5e94e80-d8d4-11eb-8d43-d9aae595985a.mp4

**Comment by goga-m** [comment]
@w3ea yeah basically the confirmation modal issue in settings is in another task https://app.clickup.com/t/n16mfa and could be a a subject of a separate pr. Thanks for the video :+1:

**Review by w3ea — APPROVED** [review]

**Review by faustbrian — APPROVED** [review]

**Pull request merged**